monitoring/README.md

```diff
@@ -1,112 +1,87 @@
 
-1. Create your hosts file with the list of target hosts (KCP server & Edge Workload Execution Clusters (WEC))
+1. Install the monitoring tools:
 
-```
-[kcp-server]
-192.168.56.2
+Starting from a local directory containing the git repo, do the following.
 
-[edge-wecs]
-192.160.56.10
-192.160.56.12
+```bash
+$ cd monitoring
+$ ./install-ks-monitoring.sh
 ```
 
-2. Configure the Prometheus targets endpoints:
+Optionally, check the deployment of the monitoring tools:
+
+```bash
+$kubectl -n ks-monitoring get pods
+NAME                                                  READY   STATUS    RESTARTS   AGE
+grafana-7969f6cf5c-lp4g6                              1/1     Running   0          66s
+prometheus-kube-prometheus-operator-b56d747c9-wfs2w   1/1     Running   0          42s
+prometheus-prometheus-kube-prometheus-prometheus-0    2/2     Running   0          39s
+pyroscope-0                                           1/1     Running   0          90s
+pyroscope-alloy-0                                     2/2     Running   0          90s
+```
 
-a) Prometheus Server: edit the file roles/prometheus/templates/prometheus-config.yaml.j2
+2. Configure Prometheus and Pyroscope to scrape your KubeStellar spaces:
 
+```bash
+$ ./configure-metrics-tools.sh
 ```
-global:
-  evaluation_interval: 5s
-  external_labels:
-    env: dev
-  scrape_interval: 30s
-scrape_configs:
-- job_name: mailbox-controller
-  scrape_interval: 15s
-  metrics_path: /metrics
-  static_configs:
-  - targets:
-    - '<host-ipAddress>:10203'
-
-- job_name: node-exporter
-  scrape_interval: 15s
-  metrics_path: /metrics
-  static_configs:
-  - targets:
-    - '<host-ipAddress>:9100'
-
-- job_name: kcp
-  scrape_interval: 15s
-  scheme: https
-  metrics_path: /metrics
-  tls_config:
-    insecure_skip_verify: true
-  static_configs:
-  - targets:
-    - '<host-ipAddress>:6443'
 
+Optionally, check the prometheus service monitor objects:
+
+```bash
+$kubectl -n ks-monitoring get servicemonitor -l "app.kubernetes.io/part-of!=kube-prometheus-stack"
+NAME                      AGE
+ks-controller-manager     70m
+ks-transport-controller   70m
+wds-apiserver             70m
 ```
 
 
-b) Prometheus Agent: edit the file roles/prometheus/templates/prometheus-agent-config.yaml.j2
+3. Connect to Grafana UI: 
 
-```
-# my global config
-global:
-  scrape_interval: 15s
-  evaluation_interval: 15s 
-scrape_configs:
-  - job_name: "prometheus-agent"
-
-    static_configs:
-      - targets: ["{{ ansible_default_ipv4.address }}:<port_number>"]
-remote_write:
-  - url: "http://<prometheus-server-ip-address>:<port_number>/api/v1/write"
+a) Port-forward Grafana to localhost, by using the kubectl command:
+
+```bash
+$kubectl port-forward -n ks-monitoring svc/grafana 3000:80
 ```
 
+b) Check Prometheus and Pyroscope data source configuration: in a browser, go to the Grafana server at http://localhost:3000 and on the left-hand side, go to Configuration > Data sources.
```

Comment:
Your picture shows that the path is actually "Home > Connections > Data sources". That matches my experience.

Reply:
Fixed!